app/javascript/components/patient/vaccines/VaccineModal.js

```diff
@@ -0,0 +1,244 @@
+            <Row>
+              <Form.Group as={Col}>
+                <Form.Label htmlFor="group-name-select" className="nav-input-label">
+                  Vaccine Group
+                </Form.Label>
+                <Select
+                  inputId="group-name-select"
+                  name="group_name"
+                  defaultValue={
+                    this.props.currentVaccineData?.group_name
+                      ? { label: this.props.currentVaccineData.group_name, value: this.props.currentVaccineData.group_name }
+                      : { label: defaultGroupNameOption, value: defaultGroupNameOption }
+                  }
+                  value={{ label: this.state.group_name, value: this.state.group_name }}
+                  options={this.props.group_name_options?.map(option => {
+                    return { label: option, value: option };
+                  })}
+                  onChange={this.handleGroupNameChange}
+                  placeholder=""
+                  theme={theme => ({
+                    ...theme,
+                    borderRadius: 0,
+                  })}
+                  styles={{ menu: base => ({ ...base, zIndex: 9999 }), option: base => ({ ...base, minHeight: 30 }) }}
+                />
+              </Form.Group>
+            </Row>
+            <Row>
+              <Form.Group as={Col}>
+                <Form.Label htmlFor="product-name-select" className="nav-input-label">
+                  Product Name
+                </Form.Label>
+                <Select
+                  inputId="product-name-select"
+                  name="product_name"
+                  defaultValue={
+                    this.props.currentVaccineData?.product_name
+                      ? { label: this.props.currentVaccineData.product_name, value: this.props.currentVaccineData.product_name }
+                      : ''
+                  }
+                  value={this.state.product_name ? { label: this.state.product_name, value: this.state.product_name } : ''}
+                  options={this.getProductNameOptions().map(option => {
+                    return { label: option, value: option };
+                  })}
+                  onChange={this.handleProductNameChange}
+                  placeholder=""
+                  theme={theme => ({
+                    ...theme,
+                    borderRadius: 0,
+                  })}
+                  styles={{ menu: base => ({ ...base, zIndex: 9999 }), option: base => ({ ...base, minHeight: 30 }) }}
+                />
+              </Form.Group>
+            </Row>
+            <Row>
+              <Form.Group as={Col}>
+                <Form.Label htmlFor="administration-date" className="nav-input-label">
+                  Administration Date
+                </Form.Label>
+                <DateInput
+                  id="administration-date"
+                  date={this.state.administration_date}
+                  minDate={'1900-01-01'}
```

Comment:
is this `minDate` this far back because of much older vaccines for other types of diseases?

Reply:
Yep! This and the max date were decided upon with the PO team (and that was the reasoning).